flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/exceptions/UnfulfillableSlotRequestException.java

```diff
@@ -31,4 +32,9 @@ public UnfulfillableSlotRequestException(PendingSlotRequest pendingSlotRequest)
 		super("Could not fulfill slot request " + pendingSlotRequest.getAllocationId() + ". "
 			+ "Requested resource profile (" + pendingSlotRequest.getResourceProfile() + ") is unfulfillable.");
 	}
+
+	@VisibleForTesting
+	public UnfulfillableSlotRequestException() {
+		super("UnfulfillableSlotRequestException for testing purpose.");
+	}
```

Comment:
Instead of introducing this constructor for tests only, I would change the existing on to accept `UnfulfillableSlotRequestException(AllocationID, ResourceProfile)`.